# Do not overrite -mcpu flag
diff -N -U 3 -r mjpegtools-1.9.0-orig/configure mjpegtools-1.9.0/configure
--- mjpegtools-1.9.0-orig/configure	2009-01-06 08:49:27.000000000 +0200
+++ mjpegtools-1.9.0/configure	2010-01-07 13:17:30.000000000 +0200
@@ -25986,7 +25986,7 @@
 echo $ECHO_N "checking sub-architecture settings... $ECHO_C" >&6; }
     if test x$have_x86cpu = xtrue; then
        host_mod_cpu=`echo $host_cpu | tr _ -`
-       ARCHFLAGS="-march=$host_mod_cpu -mcpu=$host_mod_cpu"
+       ARCHFLAGS="-march=$host_mod_cpu"
     fi
   else
     if test ! -r $srcdir/cpuinfo.sh; then
diff -N -U 3 -r mjpegtools-1.9.0-orig/configure.ac mjpegtools-1.9.0/configure.ac
--- mjpegtools-1.9.0-orig/configure.ac	2009-01-06 08:33:48.000000000 +0200
+++ mjpegtools-1.9.0/configure.ac	2010-01-07 13:18:34.000000000 +0200
@@ -432,7 +432,7 @@
     AC_MSG_CHECKING(sub-architecture settings)
     if test x$have_x86cpu = xtrue; then
        host_mod_cpu=`echo $host_cpu | tr _ -`
-       ARCHFLAGS="-march=$host_mod_cpu -mcpu=$host_mod_cpu"
+       ARCHFLAGS="-march=$host_mod_cpu"
     fi
   else
     if test ! -r $srcdir/cpuinfo.sh; then
diff -N -U 3 -r mjpegtools-1.9.0-orig/cpuinfo.sh mjpegtools-1.9.0/cpuinfo.sh
--- mjpegtools-1.9.0-orig/cpuinfo.sh	2008-11-30 08:30:23.000000000 +0200
+++ mjpegtools-1.9.0/cpuinfo.sh	2010-01-07 13:20:02.000000000 +0200
@@ -24,7 +24,7 @@
 if test $_cc_major -ge 4; then
   _opt_mcpu="-mtune"
 else
-  _opt_mcpu="-mcpu"
+  _opt_mcpu="-march"
 fi
 
 do_cc()
# Fix compile error with GCC 4.4 and above (see https://sourceforge.net/p/mjpeg/mailman/message/24922627/ )
diff -urNad mjpegtools-1.9.0-orig/mplex/lpcmstrm_in.cpp mjpegtools-1.9.0/mplex/lpcmstrm_in.cpp
--- mjpegtools-1.9.0-orig/mplex/lpcmstrm_in.cpp	2005-10-13 19:43:10.000000000 +0300
+++ mjpegtools-1.9.0/mplex/lpcmstrm_in.cpp	2016-09-17 16:35:41.000000000 +0300
@@ -53,7 +53,7 @@
 
 bool LPCMStream::Probe(IBitStream &bs )
 {
-    char *last_dot = strrchr( bs.StreamName(), '.' );
+    const char *last_dot = strrchr( bs.StreamName(), '.' );
     return 
         last_dot != NULL 
         && strcmp( last_dot+1, "lpcm") == 0;
